### PR TITLE
Do not autofollow back a user you are ignoring

### PR DESCRIPTION
**Pull request by Flaburgan**
Fixes #7808 

I also added a test for the auto follow back feature as it looks we didn't have one.

### CODE REVIEW

**Comment by jhass** [comment]
Makes me wonder, should we fail following a user at all if they're ignored, so an abstraction level below this? Would that break anything UI side?

**Comment by Flaburgan** [comment]
Yeah I asked myself the same question. I think we should not block userA for following userB, even if userB is blocking userA. diaspora* relationship is asymmetrical, if userA wants to share with userB (so make is private profile available, even limited posts available) then that's fine. UserB blocking userA can still go to userA profile sometimes and see posts there.

**Comment by jhass** [comment]
Yeah sure, it's not what I meant, rather doing the same check as this, just generally in `User#share_with`, so failing there if you try to follow a user you ignore.

**Comment by Flaburgan** [comment]
>if you try to follow a user you ignore.

The UI doesn't allow you to do that, the aspect dropdown selector becomes "Stop ignoring" when you ignore someone. But I guess it should be done in the controller anyway, as someone can manually send such a request.

**Comment by jhass** [comment]
And instead of patching all the holes  (controller, autofollowback, what else did we forget), I wonder if it makes sense to just put the filter into the one place that all these call.

**Comment by Flaburgan** [comment]
Agree.

I don't see a method `sharing_with` in the `user.rb` file though, can you please point me to the place where you want me to move the check? Thanks!

**Comment by jhass** [comment]
I was thinking of https://github.com/diaspora/diaspora/blob/develop/app/models/user/connecting.rb

File: app/models/contact.rb [hunk 1]
```diff
@@ -63,7 +63,8 @@ def in_aspect?(aspect)
 
   # Follows back if user setting is set so
   def receive(_recipient_user_ids)
-    user.share_with(person, user.auto_follow_back_aspect) if user.auto_follow_back && !receiving
+    return if !user.auto_follow_back || receiving || user.blocks.where(person_id: person.id).exists?
```
Comment (diaspora-code-review) [inline app/models/contact.rb]: Layout/EmptyLineAfterGuardClause: Add empty line after guard clause.

**Comment by Flaburgan** [comment]
@jhass PR updated to do the test in the `share_with` method.

**Review by jhass — APPROVED** [review]

**Pull request closed without merging**